app/src/main/java/org/schabi/newpipe/fragments/list/BaseListInfoFragment.java

```diff
@@ -182,15 +183,17 @@ private void allowDownwardFocusScroll() {
     @Override
     public void handleNextItems(final ListExtractor.InfoItemsPage result) {
         super.handleNextItems(result);
-        currentNextPageUrl = result.getNextPageUrl();
+        currentNextPage = result.getNextPage();
         infoListAdapter.addInfoItemList(result.getItems());
 
         showListFooter(hasMoreItems());
     }
 
     @Override
     protected boolean hasMoreItems() {
-        return !TextUtils.isEmpty(currentNextPageUrl);
+        return currentNextPage != null
+                && ((currentNextPage.getUrl() != null && !currentNextPage.getUrl().isEmpty())
+                || (currentNextPage.getIds() != null && !currentNextPage.getIds().isEmpty()));
```

Comment:
You can use `Utils.isNullOrEmpty()` from NewPipe Extractor as it's now merged.
You'll also see `TextUtils.isEmpty()` in NewPipe code, it's a method from `android.text` package, but I don't like the name because it doesn't tell about null check.